Adding protected method to convert PKCS12 key to PEM.

dhermes opened the pull request:
@craigcitro LMK if you think this should be a standalone function (possibly in `crypt.py`) instead of a protected method on `SignedJwtAssertionCredentials`.

craigcitro commented:
+1, i'd rather see it as a separate function (which i suspect could be used elsewhere).

dhermes committed:
Adding protected method to convert PKCS12 key to PEM.

dhermes commented:
@craigcitro I factored out and made it public in `oauth2client.crypt`. I'm not sure how I feel about

- Putting the tests in `test_jwt` (but there is no `test_crypt`).
- Raising an `ImportError` at runtime. I can "resolve" this by just having the function convert PKCS12 instead of trying to also handle existing PEM keys.

craigcitro commented:
* for the tests, why not just make a `test_crypt.py`?
* i agree about the `ImportError`, but i don't see an obvious clean answer. i'm fine with (1) doing either an `ImportError` or `NotImplementedError` and (2) filing an issue to deal with the openssl dependency. (ultimately it'd be nice to just drop it.)

dhermes commented:
OpenSSL is a conditional dependency, that's the whole point of crypt.py (graceful degradation).

If I tear our the PEM part, then I can define this conditionally as OpenSSLVerifier and OpenSSLSigner are done.

(I have talked myself out of the ImportError.)

craigcitro commented:
wfm

dhermes commented:
@craigcitro PTAL

dhermes commented:
@craigcitro PTAL

dhermes committed:
Moving private_key_as_pem->pkcs12_key_as_pem.

Also only defining if OpenSSL is installed and conditionally
defining a method which raises NotImplementedError if not
defined.

craigcitro commented on tests/test_crypt.py:
```diff
@@ -0,0 +1,73 @@
+
+
+def datafile(filename):
+  f = open(os.path.join(os.path.dirname(__file__), 'data', filename), 'rb')
```
why not just `pkgutil.get_data`?

dhermes replied:
This is copy pasta from `test_jwt`. Also `pkgutil.get_data` would not accomplish anything close to what this method does.

This runs from `tests/` and opens files in `tests/data/` via the relative path given by `filename`.

craigcitro replied:
pretty sure this whole function is just `pkgutil.get_data(__name__, os.path.join('data', filename))`. go ahead and give it a try before you decide it's unrelated. `;)`

not deeply important though, this code all needs some spring cleaning.

dhermes replied:
Interesting. My bad.

dhermes commented:
Thanks Craig!

In the interest of closing GoogleCloudPlatform/gcloud-python#537, when do you think a new release will be cut?

craigcitro commented:
i should be able to cut a release tomorrow.

craigcitro commented:
actually, let's make that friday -- so that i can be teaching two other folks to do it at the same time.

craigcitro commented:
new release is out!

dhermes commented:
Thanks! I noticed: https://github.com/GoogleCloudPlatform/gcloud-python/pull/559#issuecomment-70327114
